omnibus/config/software/datadog-agent-integrations-py2.rb

```diff
@@ -140,6 +140,8 @@
     # Some libraries (looking at you, aerospike-client-python) need EXT_CFLAGS instead of CFLAGS.
     nix_specific_build_env = {
       "aerospike" => nix_build_env.merge({"EXT_CFLAGS" => nix_build_env["CFLAGS"] + " -std=gnu99"}),
+      # Always build pyodbc from source to link to the embedded version of libodbc
+      "pyodbc" => nix_build_env.merge({"PIP_NO_BINARY" => "pyodbc"}),
```

Comment:
Should we also add `pyodbc` to the excluded list so that it doesn't flow in from our `agent-requirements.in` as is?

Reply:
We need to exclude them if and only if we replace the install method with a custom recipe. Here I just need to apply a flag to tell pip to build the wheel. This is the purpose of this `nix_specific_build_env`, we apply custom flag "on the fly" when reading the `agent-requirements.in` file. You end up with something like this during the build: 

```
[Builder: datadog-agent-integrations-py3] I | 2023-10-17T11:25:24+00:00 | Environment:
[Builder: datadog-agent-integrations-py3] I | 2023-10-17T11:25:24+00:00 |   CC="/opt/gcc-10.4.0/bin/gcc"
[Builder: datadog-agent-integrations-py3] I | 2023-10-17T11:25:24+00:00 |   CFLAGS="-I/opt/datadog-agent/embedded/include -I/opt/mqm/inc -std=c99"
[Builder: datadog-agent-integrations-py3] I | 2023-10-17T11:25:24+00:00 |   CXX="/opt/gcc-10.4.0/bin/g++"
[Builder: datadog-agent-integrations-py3] I | 2023-10-17T11:25:24+00:00 |   CXXFLAGS="-I/opt/datadog-agent/embedded/include -I/opt/mqm/inc"
[Builder: datadog-agent-integrations-py3] I | 2023-10-17T11:25:24+00:00 |   LDFLAGS="-L/opt/datadog-agent/embedded/lib -L/opt/mqm/lib64 -L/opt/mqm/lib"
[Builder: datadog-agent-integrations-py3] I | 2023-10-17T11:25:24+00:00 |   LD_RUN_PATH="/opt/datadog-agent/embedded/lib -L/opt/mqm/lib64 -L/opt/mqm/lib"
[Builder: datadog-agent-integrations-py3] I | 2023-10-17T11:25:24+00:00 |   PATH="/opt/datadog-agent/embedded/bin:/usr/local/rvm/gems/ruby-2.7.2/bin:/root/.cargo/bin:/usr/local/rvm/gems/ruby-2.7.2@global/bin:/usr/local/rvm/rubies/ruby-2.7.2/bin:/root/miniconda3/envs/ddpy3/bin:/root/miniconda3/condabin:~/.cargo/bin:/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/rvm/bin"
[Builder: datadog-agent-integrations-py3] I | 2023-10-17T11:25:24+00:00 |   PIP_CONFIG_FILE="/go/src/github.com/DataDog/datadog-agent/pip.conf"
[Builder: datadog-agent-integrations-py3] I | 2023-10-17T11:25:24+00:00 |   PIP_FIND_LINKS="/omnibus/src/datadog-agent-integrations-py3/integrations-core/.build_deps"
[Builder: datadog-agent-integrations-py3] I | 2023-10-17T11:25:24+00:00 |   PIP_NO_BINARY="pyodbc"
[Builder: datadog-agent-integrations-py3] I | 2023-10-17T11:25:24+00:00 | $ /opt/datadog-agent/embedded/bin/python3 -m pip install --no-deps --require-hashes -r /opt/datadog-agent/agent_pyodbc_requirements-py3.txt
[Builder: datadog-agent-integrations-py3] I | 2023-10-17T11:25:33+00:00 | Execute: `/opt/datadog-agent/embedded/bin/python3 -m pip install --no-deps --require-hashes -r /opt/datadog-agent/agent_pyodbc_requirements-py3.txt': 8.9167s
[Builder: datadog-agent-integrations-py3] I | 2023-10-17T11:25:33+00:00 | delete `/opt/datadog-agent/agent_pyodbc_requirements-py3.txt': 0.0003s
```

Reply:
Thanks for the clarification!